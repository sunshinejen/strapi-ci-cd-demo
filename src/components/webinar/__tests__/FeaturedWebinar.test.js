import React from 'react';
import {render, screen } from '@testing-library/react';
import FeaturedWebinar from '../featuredWebinar';

//Snapshot Test
const data = {
    date: 'Thursday, July 09, 2037',
    time: "3:00pm",
    webinar_title: 'Racism, Why We Must Solve It in Technology',
    webinar_description: "We explore how can technology improve the treatment of racial groups and their interactions with the Police, and criminal justice system, improve equity in education, improve financial opportunities and more.",
    flyer_img: "https://res.cloudinary.com/dlz977spb/image/upload/v1659651175/racism_solve_in_tech_flyer_bfdfe16a62.png",
    registration: "https://docs.google.com/forms/d/e/1FAIpQLSeV3qKNMJDTZ6jb1sKlesJNCU4p_x4D0r5ZUUZ3VRgHsznJ1g/viewform",
}

//Snapshot Test 
it("Renders as expected", () => {
    render(<FeaturedWebinar webinar={data} />)
});

test("Featured Webinar snapshot", () => {
    const { asFragment } = render(<FeaturedWebinar webinar={data} />)
    expect(asFragment()).toMatchSnapshot()
});

//Unit Test
test('Featured Webinar page', () => {

    const { getByText } = render(<FeaturedWebinar webinar={data} />)


    const testImage = document.querySelector("img")
    expect(testImage.alt).toContain("...")
    expect(getByText('Thursday, July 09, 2037')).toBeInTheDocument()
    expect(getByText('3:00pm')).toBeInTheDocument()
    expect(getByText('We explore how can technology improve the treatment of racial groups and their interactions with the Police, and criminal justice system, improve equity in education, improve financial opportunities and more.')).toBeInTheDocument()
    expect(screen.getByRole('link')).toHaveAttribute('href', 'https://docs.google.com/forms/d/e/1FAIpQLSeV3qKNMJDTZ6jb1sKlesJNCU4p_x4D0r5ZUUZ3VRgHsznJ1g/viewform')
});

