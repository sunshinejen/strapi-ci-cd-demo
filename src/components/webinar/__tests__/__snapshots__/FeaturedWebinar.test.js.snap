// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Featured Webinar snapshot 1`] = `
<DocumentFragment>
  <div
    class="page-section clearfix"
  >
    <div
      class="container"
    >
      <div
        class="intro"
      >
        <img
          alt="..."
          class="intro-img img-fluid mb-3 mb-lg-0 rounded"
        />
        <div
          class="intro-text left-0 bg-faded p-4 rounded"
        >
          <h6
            class="text-muted"
          >
            Thursday, July 09, 2037
          </h6>
          <h6
            class="timeSubtitle"
          >
            3:00pm
          </h6>
          <h2
            class="section-heading mb-4"
          >
            <span
              class="section-heading-lower"
            >
              <h4>
                 Racism, Why We Must Solve It in Technology
              </h4>
            </span>
          </h2>
          <div
            class="mb-3"
          >
            <p>
              We explore how can technology improve the treatment of racial groups and their interactions with the Police, and criminal justice system, improve equity in education, improve financial opportunities and more.
            </p>
          </div>
          <a
            class="button"
            data-testid="registration-link"
            href="https://docs.google.com/forms/d/e/1FAIpQLSeV3qKNMJDTZ6jb1sKlesJNCU4p_x4D0r5ZUUZ3VRgHsznJ1g/viewform"
            style="text-decoration: none;"
            target="_blank"
            type="button"
          >
            Webinar Registration
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
